app/resources/collections/collections_controller.rb

```diff
@@ -43,6 +43,22 @@ def ark_report
     end
   end
 
+  def change_set_class
+    if params[:change_set].present? || (resource_params && resource_params[:change_set].present?)
+      DynamicChangeSet.class_from_param(params[:change_set] || resource_params[:change_set])
+    else
+      DynamicChangeSet
+    end
+  end
+
+  def render_new
+    if change_set_param
+      redirect_to polymorphic_path([:new, change_set_param, self.class.resource_class])
```

Comment:
This is still a problem, but `render :new` does not preserve the state of the `@change_set`.

Reply:
I was incorrect, the value of the `change_set` and the state captured with `@change_set` are preserved, but the request parameters do not contain the ChangeSet name (which are what is parsed in https://github.com/pulibrary/figgy/blob/master/app/helpers/metadata_form_helper.rb#L11).